### PR TITLE
Update CSF script helper methods to use `curriculum_umbrella`

### PR DESCRIPTION
**Pull request by Erin007**
In #28678 I introduced a new property on scripts, `curriculum_umbrella` which can be used to track if a given script is part of one of our 3 core courses (CSF, CSD or CSP) regardless of version year. This was done in an effort to refactor all of the hardcoded lists we have to maintain and update each time we launch a new version of a course. 

This PR uses `under_curriculum_umbrella` and a revised `csf?` method to addresses 3 helper methods that used hardcoded lists of CSF versions:
[LP-443](https://codedotorg.atlassian.net/browse/LP-443) --> `has_banner?`
[LP-444](https://codedotorg.atlassian.net/browse/LP-444) --> `k5_course`
[LP-447](https://codedotorg.atlassian.net/browse/LP-447) --> `csf?`

Currently, `curriculum_umbrella == 'CSF'` for the following scripts:  ["20-hour", "course1", "course2", "course3", "course4", "coursea-2017", "coursea-2018", "courseb-2017", "courseb-2018", "coursec-2017", "coursec-2018", "coursed-2017", "coursed-2018", "coursee-2017", "coursee-2018", "coursef-2017", "coursef-2018", "express-2017", "express-2018", "pre-express-2017", "pre-express-2018", "coursed-2019", "coursee-2019", "coursef-2019", "express-2019", "coursec-2019", "coursea-2019", "courseb-2019", "pre-express-2019"]

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -826,9 +839,10 @@ def has_lesson_pdf?
 
   def has_banner?
     # Temporarily remove Course A-F banner (wrong size) - Josh L.
```
Comment (Erin007) [inline dashboard/app/models/script.rb:841]: This [temporary change has been kickin' around for almost 2 years](https://github.com/code-dot-org/code-dot-org/pull/16621). @joshlory or maybe @ryansloan do you know if it's still needed? Should we add re-sizing banners for Course A-F to the LP board?

Comment (ryansloan) [reply dashboard/app/models/script.rb:841]: Everything is temporary...
We should get banners on Mark's backlog and make sure they're correctly sized, then we can remove this temporary change. I think @fontie715 should drive, LMK if I can help!

**Comment by fontie715** [comment]
Sounds good - tracked by https://codedotorg.atlassian.net/browse/LP-452

File: dashboard/app/models/script.rb [hunk 2]
```diff
@@ -676,16 +676,29 @@ def csf_international?
     ScriptConstants.script_in_category?(:csf_international, name)
   end
 
+  def self.script_names_by_curriculum_umbrella(curriculum_umbrella)
+    Script.where("properties -> '$.curriculum_umbrella' = ?", curriculum_umbrella).pluck(:name)
+  end
+
+  def under_curriculum_umbrella(specific_curriculum_umbrella)
```
Comment (maddiedierker) [inline dashboard/app/models/script.rb]: nit: rename to `under_curriculum_umbrella?` since the return value is a boolean

**Review by maddiedierker — APPROVED** [review]

**Pull request merged**